### DISCUSSION TITLE
directus_ files id cannot use UUID 5

### DISCUSSION THREAD

**Discussion by cdwmhcc** [directus/directus#8994] [Archive]
I am migrating the database. During the migration, I found that uuid5 is used as directus_files after generates the ID, the files/photos cannot be loaded.
If I use uuid4, I occasionally generate duplicate IDS, resulting in migration failure.

If I create my own collection and use uuid5, everything is normal.
How should I be using uuid5 in directus_files .

**Comment by rijkvanzanten** [comment]
Directus relies on UUIDv4 for its UUID primary keys. There's no official way to change this

**Reply by cdwmhcc** [reply]
Oh,This is bad news. The uuid4 generated during the data migration process is often repeated. I am using: https://www.npmjs.com/package/uuid

**Reply by benhaynes** [reply]
I'm very curious how many items are being added for the UUID4 key to be duplicated... it is still a universally unique identifier, so conflicts seem so unlikely.

**Reply by cdwmhcc** [reply]
> I'm very curious how many items are being added for the UUID4 key to be duplicated... it is still a universally unique identifier, so conflicts seem so unlikely.

The data is not too much, only a few hundred thousand in total. I am also curious why it repeats, it happens occasionally, but not every time.
